Various suggestions for the docs (#7423)
netdata/netdata: database/engine/README.md

```

The above values are the default and minimum values for Page Cache size and DB engine disk space quota. Both numbers are

> in **MiB**. **Each DB engine instance will allocate it's own page cache size and disk space**, thus for N instances, DBengine will need N * Resources.

Reviewer: ```suggestion
in **MiB**. **Each DB engine instance will allocate it's own page cache size and disk space**, thus for N instances, DB engine will need `N * resources`.
```

The `page cache size` option determines the amount of RAM in **MiB** that is dedicated to caching Netdata metric values
themselves, as far as queries are concerned. The total page cache size will be greater since data collection itself will
consume additional memory as is described in the [Memory requirements](#memory-requirements) section.

The `dbengine disk space` option determines the amount of disk space in **MiB** that is dedicated to storing Netdata
metric values and all related metadata describing them.

> ### DBengine storage capacity in Metrics

Reviewer: ```suggestion
### DB engine storage capacity
```

An important observation is that RAM usage depends on both the `page cache size` and the `dbengine disk space` options.

> ### Example:

Reviewer: Let's come up with a header for this that's more descriptive than "example."

Let's say we have 1 Netdata Master Node with 10 slaves. I want to save the data for 1 week without downsampling, and afterwards I will downsample to 1 sample per 5 minutes.

For the archiving I will be using the Backends functionality, so we wont't elaborate any further here.

For the one week's worth of data, I will have 10 DB engine instances for the *Slaves* and 1 for the *Master*.

Each Slave saves 2000 metrics/s, and Master Node only 1000/s.

Thus, taking into account that for 4000 metrics/s the default settings will save 1 day, with 2000 metrics/s I will save for 2 days.

Final Disk Space = (256 * 10 slaves * 7 days ) / 2 + (256 * 7 * 1 Master) / 4 MiB

> Final RAM usage = (10 slaves + 1 Master)* (32 * 1.049 * 10^6) Page cache size + (2000 dimensions * 10 slaves * 4096 * 2 ) + (#pages-on-disk * 4096 * 0.03) + (1000 dimensions * 1 Master * 4096 * 2 ) + (#pages-on-disk * 4096 * 0.03) bytes

Reviewer: A few comments on this new section:

We don't use "I" in our documentation—these pronouns should be changed to "you," as though the reader is the one setting up this configuration and trying to understand memory usage.

I wouldn't capitalize words like _Slaves_, _Master_, or _Node_.

I'd like to see some example numbers from those last two calculations. Asking the reader to get out a calculator to do it themselves is too complex. I also wonder if there's a way to present the numbers in a clearer fashion. Even putting them into a code block would help make it more readable.

Author: I will make it more readable. 

Since we are discussing about this, can we see how to compute the variable #pages-on-disk, since I was unable to find it. Moreover, as a user, I would like to read about the correlation between `disk space`-`page cache size` and metrics retention period.

At the moment, I can only proceed with rule of 3 (i.e if I want double the metrics, I need to double both variables for the same retention period)

## File descriptor requirements

The Database Engine may keep a **significant** amount of files open per instance (e.g. per streaming slave or master
server). When configuring your system you should make sure there are at least 50 file descriptors available per
`dbengine` instance.

> *Netdata allocates 25% of the available file descriptors to its Database Engine instances.* This means that only 25% of

Reviewer: ```suggestion
*Netdata allocates 25% of the available file descriptors to its db engine instances.* This means that only 25% of
```

> the file descriptors that are available to the Netdata service are accessible by dbengine instances. 

Reviewer: ```suggestion
the file descriptors that are available to the Netdata service are accessible by db engine instances. 
```
